### PR TITLE
Modify JSR166TestCase to gather more error information

### PR DESCRIPTION
**Pull request by theresa-m**
- Update Xdump to catch JUnit exception and AssertionFailedError

### CODE REVIEW

File: test/jdk/java/util/concurrent/tck/JSR166TestCase.java [hunk 1]
```diff
@@ -431,7 +437,7 @@ protected void printFooter(TestResult result) {
      */
     static junit.textui.TestRunner newPithyTestRunner() {
         junit.textui.TestRunner runner = new junit.textui.TestRunner();
-        runner.setPrinter(new PithyResultPrinter(System.out));
+        //runner.setPrinter(new PithyResultPrinter(System.out));
```
Comment (keithc-ca) [inline test/jdk/java/util/concurrent/tck/JSR166TestCase.java:440]: Please explain this change.

Comment (theresa-m) [reply test/jdk/java/util/concurrent/tck/JSR166TestCase.java:440]: I thought based on the `newPithyTestRunner ` comment there might be more information printed when running tests but I tried it by forcing one of the tests to fail locally and that doesn't really seem to be the case, I'll revert this.

**Comment by keithc-ca** [comment]
Jenkins test sanity.openjdk win jdknext

**Comment by pshipton** [comment]
There are no working Windows build machines.

**Comment by keithc-ca** [comment]
Jenkins test sanity.openjdk zlinux jdknext

**Comment by theresa-m** [comment]
The test failures are not related to this change, I think they are due to in progress jdk 24 support.

**Review by keithc-ca — APPROVED** [review]

**Review by keithc-ca — CHANGES_REQUESTED** [review]
Please update the commit message and description here to remove the reference to changing the TestRunner (which was reverted).

**Review by keithc-ca — APPROVED** [review]

**Comment by theresa-m** [comment]
I opened prs to port this change to JDK 23, 21, 17, 11. Can you please review these as well @keithc-ca ?
* https://github.com/ibmruntimes/openj9-openjdk-jdk23/pull/82
* https://github.com/ibmruntimes/openj9-openjdk-jdk21/pull/231
* https://github.com/ibmruntimes/openj9-openjdk-jdk17/pull/418
*  https://github.com/ibmruntimes/openj9-openjdk-jdk11/pull/850

**Pull request merged**